src/utils/api-request.ts

```diff
@@ -178,12 +178,32 @@ export class HttpClient {
  * Sends an HTTP request based on the provided configuration. This is a wrapper around the http and https
  * packages of Node.js, providing content processing, timeouts and error handling.
  */
-function sendRequest(config: HttpRequestConfig): Promise<LowLevelResponse> {
+function sendRequest(httpRequestConfig: HttpRequestConfig): Promise<LowLevelResponse> {
+  const config: HttpRequestConfig = deepCopy(httpRequestConfig);
   return new Promise((resolve, reject) => {
     let data: Buffer;
     const headers = config.headers || {};
     if (config.data) {
-      if (validator.isObject(config.data)) {
+      // GET and HEAD do not support body in request.
```

Comment:
Here's what I'd suggest we do here:

1. Introduce a new `queryParams: any` field to `HttpRequestConfig`.
2. Append all key-value items in `queryParams` to the URL, for all requests.

```
let url: string;
if (config.queryParams) {
  url = <config.url + config.queryParams>
} else {
  url = config.url;
}
```

3. If a GET or HEAD request has been made with a `data` value, throw an error.

This way we don't have to mutate the `config` object in flight.

Reply:
Done